students/johnpharmd/lesson02/series.py

```diff
@@ -0,0 +1,91 @@
+def fibonacci(n):
+    """
+    Given integer parameter n, this function returns the nth value
+    in the Fibonacci series. This series is recursive, that is,
+    each value in the series is the result of summing the two values
+    immediately before it. The series moves forward, so to speak,
+    by summing backward. Computationally, this is accomplished by
+    calling the function from within the function, which leads
+    to more function calls until a result is generated--recursion.
+    """
+
+    if n == 1:
+        v = 0
+    elif n == 2:
+        v = 1
+    else:
+        v = fibonacci(n - 2) + fibonacci(n - 1)
+
+    return v
+
+
+def lucas(n):
+    """
+    Given integer parameter n, this function returns the nth value
+    in the Lucas series. Like the Fibonacci series, this series
+    is recursive.
+    """
+
+    if n == 1:
+        v = 2
+    elif n == 2:
+        v = 1
+    else:
+        v = lucas(n - 2) + lucas(n - 1)
+
+    return v
+    
+
+def sum_series(i, v1=0, v2=1):
+    """
+    This function also recursively returns a series element. Given
+    integer parameter i (for index) plus default v1 and v2 values,
+    it returns the Fibonacci series value at i. The Lucas series
+    value at i can be obtained by passing v1=2 when calling the
+    function. Other series will result if different integers are
+    passed to v1 and/or v2.
+    """
+    if i == 1:
+        v = v1
+    elif i == 2: 
+        v = v2
+    else: 
+        v = sum_series(i - 2, v1, v2) + sum_series(i - 1, v1, v2)
+    return v
+
+
+# Tests
+# In the following lines, each of the series functions will be
+# tested, respectively.
+
+if __name__ != "__main__":
```

Comment:
This should be `if __name__ == "__main__"` if you want to execute this code block when running the script from the command line.